library/core/src/intrinsics.rs

```diff
@@ -1907,6 +1908,11 @@ extern "rust-intrinsic" {
     /// Allocate at compile time. Should not be called at runtime.
     #[rustc_const_unstable(feature = "const_heap", issue = "79597")]
     pub fn const_allocate(size: usize, align: usize) -> *mut u8;
+
+    /// Implementation detail of the `const_panic` feature.
```

Comment:
This might still be a good place to explain what is going on. In particular, somewhere it should explain that the `None` case is unreachable.